lazyflow/operators/ioOperators/opStreamingH5N5Reader.py

```diff
@@ -22,21 +22,68 @@
 # Python
 import logging
 import time
-import numpy
+from typing import Union
+
 import vigra
 import h5py
 import z5py
-import json
 import os
-import numpy as np
 
 from lazyflow.graph import Operator, InputSlot, OutputSlot
 from lazyflow.utility import Timer
 from lazyflow.utility.helpers import get_default_axisordering, bigintprod
+from lazyflow.utility.io_util.OMEZarrStore import get_axistags_from_spec as get_ome_zarr_axistags
 
 logger = logging.getLogger(__name__)
 
 
+def _find_or_infer_axistags(file: Union[h5py.File, z5py.N5File, z5py.ZarrFile], internalPath: str) -> vigra.AxisTags:
+    assert internalPath in file, "Existence of dataset must be checked earlier"
+    try:
+        # Look for ilastik-style axistags property.
+        axistagsJson = file[internalPath].attrs["axistags"]
+        axistags = vigra.AxisTags.fromJSON(axistagsJson)
+        axisorder = "".join(tag.key for tag in axistags)
+        if "?" in axisorder:
+            raise KeyError("?")
+        return axistags
+    except KeyError:
+        pass
+
+    try:
+        # Look for OME-Zarr metadata (found at store root, not in dataset)
+        # OME-Zarr stores with more than one multiscale don't exist in public, but the spec allows it
+        multiscale_index = None
+        for i, scale in enumerate(file.attrs["multiscales"]):
+            if any(d.get("path", "") == internalPath.lstrip("/") for d in scale.get("datasets", [])):
+                multiscale_index = i
+        if multiscale_index is None:
+            raise KeyError("no spec for dataset path")
```

Comment:
TIL about Python's `for` statements having an else clause 😅 Googling suggests that it's not just me who finds this quite obscure, so I'd rather avoid using it (particularly in a context where it can be easily misread as belonging to an `if` clause).